presto-main/src/main/java/io/prestosql/operator/window/FirstValueFunction.java

```diff
@@ -27,11 +27,21 @@
 {
     private final int argumentChannel;
     private final boolean ignoreNulls;
+    private int lastNonNull;
+
+    @Override
+    public void reset()
+    {
+        lastNonNull = -1;
+    }
 
     public FirstValueFunction(List<Integer> argumentChannels, boolean ignoreNulls)
     {
         this.argumentChannel = getOnlyElement(argumentChannels);
         this.ignoreNulls = ignoreNulls;
+        if (this.ignoreNulls) {
+            this.lastNonNull = -1;
```

Comment:
reset() is unconditional, why do we have condition here?

maybe just make this field's initial value?